frappe/public/js/frappe/utils/diffview.js

```diff
@@ -0,0 +1,110 @@
+frappe.provide("frappe.ui");
+
+frappe.ui.DiffView = class DiffView {
+	constructor(doctype, fieldname, docname) {
+		this.dialog = null;
+		this.handler = null;
+		this.doctype = doctype;
+		this.fieldname = fieldname;
+		this.docname = docname;
+
+		this.dialog = this.make_dialog();
+		this.dialog.show();
+	}
+
+	make_dialog() {
+		const get_query = () => ({
+			query: "frappe.utils.diff.version_query",
+			filters: { docname: this.docname, ref_doctype: this.doctype },
+		});
+		const onchange = () => this.compute_diff();
+		let dialog = new frappe.ui.Dialog({
+			title: __("Compare Versions"),
+			fields: [
+				{
+					label: __("From version"),
+					fieldtype: "Link",
+					fieldname: "from_version",
+					options: "Version",
+					reqd: 1,
+					get_query,
+					onchange,
+				},
+				{
+					fieldtype: "Column Break",
+					fieldname: "cb",
+				},
+				{
+					label: __("To version"),
+					fieldtype: "Link",
+					fieldname: "to_version",
+					options: "Version",
+					reqd: 1,
+					get_query,
+					onchange,
+				},
+				{
+					fieldtype: "Section Break",
+					fieldname: "sb",
+				},
+				{
+					label: __("Diff"),
```

Comment:
Since this loads via ajax can we add a default/ empty state?
some variant of "choose the versions to see the diff"
Or maybe hide the control when it has no content

Reply:
updated